SudoScript.Core/Parser.cs

```diff
@@ -119,22 +119,24 @@ private static List<ParameterNode> ParseParameters(TokenStream stream)
         List<ParameterNode> paramChildren = new List<ParameterNode>();
         
         // While leftBrace is not read, it will keep reading parameters. 
-        while (stream.Peek(true ,out Token? endToken) && endToken.Type != TokenType.LeftBrace) 
+        while (stream.Peek(true ,out Token? endToken) && endToken.Type != TokenType.LeftBrace)
         {
-            if (!stream.Expect(TokenType.Space, out Token _))
+            if (stream.Expect(TokenType.Space, out Token? _))
             {
-                throw new Exception("Parameter are expected to be separated by space");
-            }
+                // Checks whether the parameter is an identifier or a cell and parses them acordingly
+                if (stream.Peek(true, out Token? leftParenthesis) && leftParenthesis.Type == TokenType.LeftParenthesis)
+                {
+                    paramChildren.Add(ParseParameterCell(stream));
+                }
 
-            // Checks whether the parameter is an identifier or a cell and parses them acordingly
-            if (stream.Peek(true, out Token? leftParenthesis) && leftParenthesis.Type == TokenType.LeftParenthesis)
-            {
-                paramChildren.Add(ParseParameterCell(stream));
+                if (stream.Peek(true, out Token? identifier) && identifier.Type == TokenType.Identifier)
+                {
+                    paramChildren.Add(ParseParameterIdentifier(stream));
+                }
             }
-
-            if (stream.Peek(true, out Token? identifier) && identifier.Type == TokenType.Identifier)
+            else
             {
-                paramChildren.Add(ParseParameterIdentifier(stream));
+                throw new Exception("Parameter are expected to be separated by space");
             }
```

Comment:
Not a fan of this change. Before it was an early return as a validation check. Now you have introduced an extra level of nesting across the entire function. It just reads worse and is less managable Imo.